tt_metal/api/tt-metalium/fabric_types.hpp

```diff
@@ -4,6 +4,33 @@
 
 #pragma once
 
+namespace tt::tt_fabric {
+#define FABRIC_MODE_UNDEFINED 0x0000
+#define FABRIC_MODE_1D 0x0001
+#define FABRIC_MODE_2D 0x0002
+#define FABRIC_MODE_3D 0x0004
+#define FABRIC_MODE_LINE 0x0008
+#define FABRIC_MODE_RING 0x0010
+#define FABRIC_MODE_MESH 0x0020
+#define FABRIC_MODE_TORUS 0x0040
+#define FABRIC_MODE_LOW_LATENCY 0x0080
+#define FABRIC_MODE_DYNAMIC 0x0100
+
+#define FABRIC_MODE_1D_LINE (FABRIC_MODE_1D | FABRIC_MODE_LINE)
+#define FABRIC_MODE_1D_RING (FABRIC_MODE_1D | FABRIC_MODE_RING)
+#define FABRIC_MODE_1D_LINE_LOW_LATENCY (FABRIC_MODE_1D_LINE | FABRIC_MODE_LOW_LATENCY)
+#define FABRIC_MODE_1D_RING_LOW_LATENCY (FABRIC_MODE_1D_RING | FABRIC_MODE_LOW_LATENCY)
+#define FABRIC_MODE_2D_MESH (FABRIC_MODE_2D | FABRIC_MODE_MESH)
+#define FABRIC_MODE_2D_TORUS (FABRIC_MODE_2D | FABRIC_MODE_TORUS)
+#define FABRIC_MODE_2D_MESH_LOW_LATENCY (FABRIC_MODE_2D_MESH | FABRIC_MODE_LOW_LATENCY)
+#define FABRIC_MODE_2D_TORUS_LOW_LATENCY (FABRIC_MODE_2D_TORUS | FABRIC_MODE_LOW_LATENCY)
+#define FABRIC_MODE_2D_MESH_DYNAMIC (FABRIC_MODE_2D_MESH | FABRIC_MODE_DYNAMIC)
+#define FABRIC_MODE_2D_TORUS_DYNAMIC (FABRIC_MODE_2D_TORUS | FABRIC_MODE_DYNAMIC)
+
+// type for host to refer FABRIC_MODE macros in fabric_types.hpp
+using FABRIC_MODE_ = std::uint16_t;
```

Comment:
IMO this should be an enum so we can get some added type safety. Even if it's slightly redundant to the macros.

```suggestion
enum class FabricMode {
  FabricMode1DLine = FABRIC_MODE_1D_LINE,
  FabricMode1DRing = FABRIC_MODE_1D_RING,
  FabricMode1dLineLowLatency = FABRIC_MODE_1D_LINE_LOW_LATENCY,
  FabricMode1DRingLowLatency = FABRIC_MODE_1D_RING_LOW_LATENCY,
  FabricMode2DMesh = FABRIC_MODE_2D_MESH,
  FabricMode2DTorus = FABRIC_MODE_2D_TORUS,
  FabricMode2DMeshLowLatency = FABRIC_MODE_2D_MESH_LOW_LATENCY,
  FabricMode2DTorusLowLatency = FABRIC_MODE_2D_TORUS_LOW_LATENCY,
  FabricMode2DMeshDynamic = FABRIC_MODE_2D_MESH_DYNAMIC,
  FabricMode2DTorusDynamic = FABRIC_MODE_2D_TORUS_DYNAMIC
};
```